clang/test/Frontend/diags-interesting-source-region-colors.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: not %clang_cc1 %s -fmessage-length=40 -fcolor-diagnostics -fno-show-source-location -Wunused-value -o - 2>&1 | FileCheck %s
+
+// REQUIRES: ansi-escape-sequences
+
```

Comment:
I think the test here should also check that we position the caret and any source ranges correctly in the presence of front and/or back ellipses.